admiral/pkg/controller/admiral/globaltraffic_test.go

```diff
@@ -74,28 +76,217 @@ func TestGlobalTrafficAddUpdateDelete(t *testing.T) {
 
 }
 
-func TestGlobalTrafficGetByLabel(t *testing.T) {
-	config, err := clientcmd.BuildConfigFromFlags("", "../../test/resources/admins@fake-cluster.k8s.local")
-	if err != nil {
-		t.Errorf("%v", err)
+func TestGlobalTrafficController_Updated(t *testing.T) {
+
+	gth := test.MockGlobalTrafficHandler{}
+	cache := gtpCache{
+		cache: make(map[string]map[string]map[string]*v1.GlobalTrafficPolicy),
+		mutex: &sync.Mutex{},
 	}
-	stop := make(chan struct{})
-	handler := test.MockGlobalTrafficHandler{}
+	gtpController := GlobalTrafficController{
+		GlobalTrafficHandler: &gth,
+		Cache:             &cache,
+	}
+	gtp := v1.GlobalTrafficPolicy{ObjectMeta: v12.ObjectMeta{Name: "gtp", Namespace: "namespace1"}, Spec: model.GlobalTrafficPolicy{
+		Policy: []*model.TrafficPolicy {{DnsPrefix: "hello"}},
+	},}
+	gtp.Labels = map[string]string{"identity": "id", "admiral.io/env": "stage"}
+	gtpUpdated := v1.GlobalTrafficPolicy{ObjectMeta: v12.ObjectMeta{Name: "gtp", Namespace: "namespace1"}, Spec: model.GlobalTrafficPolicy{
+		Policy: []*model.TrafficPolicy {{DnsPrefix: "helloUpdated"}},
+	},}
+	gtpUpdated.Labels = map[string]string{"identity": "id", "admiral.io/env": "stage"}
+	gtpUpdatedToIgnore := v1.GlobalTrafficPolicy{ObjectMeta: v12.ObjectMeta{Name: "gtp", Namespace: "namespace1"}}
+	gtpUpdatedToIgnore.Labels = map[string]string{"identity": "id", "admiral.io/env": "stage"}
+	gtpUpdatedToIgnore.Annotations = map[string]string{"admiral.io/ignore": "true"}
 
-	globalTrafficController, err := NewGlobalTrafficController(stop, &handler, config, time.Duration(1000))
+	gtp2 := v1.GlobalTrafficPolicy{ObjectMeta: v12.ObjectMeta{Name: "gtp2", Namespace: "namespace1"}}
+	gtp2.Labels = map[string]string{"identity": "id", "admiral.io/env": "stage"}
 
-	if err != nil {
-		t.Errorf("Unexpected err %v", err)
+	//add the base object to cache
+	gtpController.Added(&gtp)
+
+	testCases := []struct {
+		name                  string
+		gtp            		  *v1.GlobalTrafficPolicy
+		expectedGtps 		  []*v1.GlobalTrafficPolicy
+	}{
+		{
+			name:                  	"Gtp with should be updated",
+			gtp:            		&gtpUpdated,
+			expectedGtps: 			[]*v1.GlobalTrafficPolicy{&gtpUpdated},
+		},
+		{
+			name:                  	"Should remove gtp from cache when update with Ignore annotation",
+			gtp:            		&gtpUpdatedToIgnore,
+			expectedGtps: 			[]*v1.GlobalTrafficPolicy{},
+		},
+	}
+	for _, c := range testCases {
+		t.Run(c.name, func(t *testing.T) {
+			gtpController.Updated(c.gtp, gtp)
+			gtpKey := common.GetGtpKey(c.gtp)
+			matchedGtps := gtpController.Cache.Get(gtpKey, c.gtp.Namespace)
+			if len(matchedGtps) != len(c.expectedGtps) {
```

Comment:
Can we use `!reflect.DeepEqual` here ?

Reply:
Similar comment for `TestGlobalTrafficController_Added` and `TestGlobalTrafficController_Deleted`